src/crab/ebpf_domain.cpp

```diff
@@ -360,7 +360,7 @@ int ebpf_domain_t::get_instruction_count_upper_bound() {
     return (ub.is_finite() && ub.number().value().fits_sint()) ? (int)ub.number().value() : INT_MAX;
 }
 
-NumAbsDomain ebpf_domain_t::check_access_packet(NumAbsDomain inv, const linear_expression_t& lb, const linear_expression_t& ub, const std::string& s,
+NumAbsDomain&& ebpf_domain_t::check_access_packet(NumAbsDomain&& inv, const linear_expression_t& lb, const linear_expression_t& ub, const std::string& s,
```

Comment:
The use of rvalue reference here is shady at best; it declares no guarantees for the state after the call, but we rely on this state later in the singleton case. It might be better to use the returned value, or rely on fast construction as before.

`NumAbsDomain&` makes it less easy to use with `when()`.

`const NumAbsDomain&` is not an option since (a) SplitDBM is never _really_ a constant due to normalization and (b) the `--assume-assert` flag require externally visible mutation.